### DISCUSSION TITLE
apng support for streamlined animation

### DISCUSSION THREAD

**Discussion by autofly17** [SuperTux/supertux#2468] [Ideas]
Currently the only way of having animated sprites/tiles is to have multiple png files for each frame, which is well...

![Screenshot](https://github.com/SuperTux/supertux/assets/130930373/d34be6e6-e05f-4aa9-a7c2-fbf295455c29)

My proposal is to add animated png support to the engine and add an `apng` symbol to the spec that dissects the file along with an optional start/end parameter, for example:

```sexp
(action
  (name "small-walk-right")
  (fps 15.0) ;; optional, default to file's fps
  (hitbox 16 26 32 32)
  (images 
    (apng "small/walk.png" 1 7) ;; 1 & 7 are optional start/end frames
  )
)
```

**Comment by HybridDog** [comment]
I would prefer the addition of JPEG XL support, including animated JPEG XL images.
JPEG XL has better lossless compression than PNG in general, is backwards-compatible to JPEG (no generation loss), has better lossy compression than JPEG, and is fast and standardized.
